src/persist-client/src/read.rs

```diff
@@ -752,6 +752,15 @@ where
     ) -> Result<Vec<LeasedBatchPart<T>>, Since<T>> {
         let batches = self.machine.snapshot(&as_of).await?;
 
+        if !PartialOrder::less_equal(self.since(), &as_of) {
+            error!(
+                "ReadHandle::snapshot called with as_of {:?} not past the since {:?}, \
+                though the read has succeeded. This means the since is not being held back correctly.",
+                &as_of,
+                self.since()
+            )
+        }
```

Comment:
subjective so feel free to ignore, but `soft_assert_or_log` might be more likely to catch these in tests.

Reply:
Done!